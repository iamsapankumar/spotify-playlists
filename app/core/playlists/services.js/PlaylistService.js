import { Component } from 'react';
import PropTypes from 'prop-types';

const noop = () => {};

export default class PlaylistService extends Component {
  static defaultProps = {
    url: '',
    onError: noop,
  };

  constructor(props) {
    super(props);
    this.state = {
    //   loading: false,
    //   data: [],
    //   error: false,
    //   response: {},
    };
  }

  fetchPlaylists = () => {
    const { onError } = this.props;
    // try {
    //   console.log('fetching playlists');
    // } catch (e) {
    //   onError(e);
    //   return e;
    // }
    return true;
  }

  render() {
    const { children } = this.props;
    const { response, error, loading } = this.state;
    const { fetchPlaylists } = this;

    return children({
      response, error, loading, fetchPlaylists,
    });
  }
}

PlaylistService.propTypes = {
  url: PropTypes.string,
  onError: PropTypes.func,
  children: PropTypes.func,
};
PlaylistService.defaultProps = {
  url: '',
  onError: noop,
  children: noop,
};
