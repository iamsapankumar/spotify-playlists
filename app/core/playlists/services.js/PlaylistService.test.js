/* eslint-disable no-undef */
import 'react-native';
import React from 'react';
// import Adapter from 'enzyme-adapter-react-16';
// import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import PlaylistService from './PlaylistService';


describe('PlaylistService', () => {
  describe('Rendering', () => {
    it('should match to snapshot =>', () => {
      const component = shallow(<PlaylistService />);
      expect(component).toMatchSnapshot();
    });
  });
  describe('Functions', () => {
    test('fetch playlist function', () => {
      const wrapper = Enzyme.shallow(<PlaylistService />);
      const action = wrapper.instance().fetchPlaylists();
      expect(action).toEqual(true);
    });
  });
});
