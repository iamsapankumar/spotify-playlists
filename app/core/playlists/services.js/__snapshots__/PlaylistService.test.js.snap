// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaylistService Rendering should match to snapshot => 1`] = `ShallowWrapper {}`;
